src/sentry/spans/consumers/process/factory.py

```diff
@@ -167,6 +169,13 @@ def _expand_segments(context_dict: dict[int, ProduceSegmentContext]):
                             continue
 
                         payload_data = prepare_buffered_segment_payload(segment)
+                        if len(payload_data) > MAX_PAYLOAD_SIZE:
+                            logger.warning(
+                                "Failed to produce message: max payload size exceeded.",
+                            )
+                            metrics.incr("performance.buffered_segments.max_payload_size_exceeded")
```

Comment:
Would it be helpful to log/tag the platform or some meta data to help narrow down where this can happen and understand if it's an actual issue (think poorly named spans from some sdk) vs just large payloads?

Reply:
yup, I ended up logging segment key!